internal/objects/reconciler_test.go

```diff
@@ -411,6 +411,27 @@ var _ = Describe("Basic propagation", func() {
 		Eventually(HasObject(ctx, "configmaps", barName, "kube-root-ca.crt")).Should(BeFalse())
 	})
 
+	It("should not propagate builtin exclusions by labels", func() {
+		SetParent(ctx, barName, fooName)
+		MakeObjectWithLabel(ctx, "roles", fooName, "role-with-labels-blocked", map[string]string{"cattle.io/creator": "norman"})
+		MakeObjectWithLabel(ctx, "roles", fooName, "role-with-labels-wrong-value", map[string]string{"cattle.io/creator": "testme"})
+		MakeObjectWithLabel(ctx, "roles", fooName, "role-with-labels-something", map[string]string{"app": "hello-world"})
+		AddToHNCConfig(ctx, api.RBACGroup, api.RoleKind, api.Propagate)
+
+		// the first one should not propagate, everything else should
+		Eventually(HasObject(ctx, "roles", barName, "role-with-labels-blocked")).Should(BeFalse())
```

Comment:
This should be `Consistently()`, not `Eventually()`. If HNC's broken and the object _is_ propagated by mistake, this line could happen to run before HNC manages to propagate it, so the test would pass incorrectly. By using `Consistently()`, HNC will keep retrying this for a few seconds (I think we've configured it as 4s), which greatly reduces the chance that the test would mistakenly pass due to a race condition.

Alternatively, you could leave this as `Eventually` but move it down by two lines, after we _know_ that at least two other objects have been propagated. That also makes it far less likely that the test passes by mistake, and will get rid of the extra 4s wait. I'm ok with either.